### PR TITLE
feat(alert): add inline variation demo

### LINKED ISSUE #3282: Inline alert demo
Create a demo that shows how to display errors and alerts within a form. Should address the following.

    Associated with the form label versus the control (displaying above the inputs/controls)
    Associated with a group of controls (eg, a group of inline checkbox controls)
    An inline alert message - at the top of the form and/or the top of the page (underneath the header and description)

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #3282

### CODE REVIEW

**Review by mcarrano — CHANGES_REQUESTED** [review]
So I think that what is missing here is showing an alert below the label for the checkbox group. Can you confirm @mmenestr ?

**Comment by mmenestr** [comment]
@christiemolloy thank you for doing this!!
@mcarrano is right, need to add an error state for the "Your experience" and "How can we contact you fields" (new pattern)!
I realize now that I didn't put any spacers or anything, so I just updated the mocks with that info!

Also small thing that's also my fault from the mocks that (which i've now updated), could you just make the spacing between the inline alert and the start of the horizontal form 24px instead of 48px? 

Otherwise looks perfect!

**Comment by mcoker** [comment]
If we want to enforce this specific spacing between the inline alert and the form, I don't think we want to use a generic layout. I'm wondering if this inline alert should be an element of the form component. Then we can add the appropriate amount of spacing per form layout modifier.

And in the stacked demo, we're using the grid layout. Is that to impose a width? Seems like we might want to have a modifier to impose that width instead if that's what we're going for.

**Review by mattnolting — CHANGES_REQUESTED** [review]
Looks like you can get rid of the layouts entirely and add `helper text` to last group..

**Comment by christiemolloy** [comment]
@mcarrano @mmenestr @mcoker @mattnolting updated do you mind re-reviewing ? 

Screen Shot 2020-09-30 at 1 56 51 PM
Screen Shot 2020-09-30 at 1 56 55 PM

I originally omitted the helper text at the bottom because the spacing is off. Should I include it inside of the `pf-c-form-group-label` so that `pf-c-form-label` and `pf-c-form-helper-text` are siblings? and then write CSS for when they are next to each other?

File: src/patternfly/components/Form/form.scss [hunk 1]
```diff
@@ -7,6 +7,9 @@
   --pf-c-form--m-horizontal__group-label--md--GridColumnGap: var(--pf-global--spacer--md);
   --pf-c-form--m-horizontal__group-control--md--GridColumnWidth: 1fr;
 
+  // limit width
+  --pf-c-form--m-limit-width--md--Width: #{pf-size-prem(500px)};
```
Comment (mattnolting) [inline src/patternfly/components/Form/form.scss:11]: Should this be `--pf-c-form--MaxWidth: auto`? Then `--pf-c-form--MaxWidth` can be updated inline on the parent `form`.

File: src/patternfly/demos/Alert/examples/horizontal-form.hbs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+              Submit form
+            {{/button}}
+            {{#> button button--modifier="pf-m-secondary" button--IsReset="true"}}
+              Cancel
+            {{/button}}
+          {{/form-actions}}
+        {{/form-group-control}}
+      {{/form-group}}
+  {{/form}}
+{{/page-main-section}}
```
Comment (mattnolting) [inline src/patternfly/demos/Alert/examples/horizontal-form.hbs:106]:
```suggestion
{{/page-main-section}}

```

**Comment by mmenestr** [comment]
Just looking at the preview site, and not the screenshot, there's a little too much spacing in the stacked form demo is. Should be no space between the label and error, and only 4px between the error and the checkboxes, instead of the 8px you have right now!

File: src/patternfly/demos/Alert/examples/stacked-form.hbs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          {{/button}}
+          {{#> button button--modifier="pf-m-secondary" button--IsReset="true"}}
+            Cancel
+          {{/button}}
+        {{/form-actions}}
+      {{/form-group-control}}
+    {{/form-group}}
+  {{/form}}
+
+{{/page-main-section}}
```
Comment (mattnolting) [inline src/patternfly/demos/Alert/examples/stacked-form.hbs:114]:
```suggestion
{{/page-main-section}}

```

File: src/patternfly/demos/Alert/examples/page-default.hbs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            {{#> nav-link nav-link--href="#"}}
+              Tertiary nav item 5
+            {{/nav-link}}
+          {{/nav-item}}
+        {{/nav-list}}
+      {{/nav}}
+    {{/page-main-nav}}
+    {{> @partial-block}}
+  {{/page-main}}
+{{/page}}
```
Comment (mattnolting) [inline src/patternfly/demos/Alert/examples/page-default.hbs:87]:
```suggestion
{{/page}}

```

File: src/patternfly/components/Form/form.scss [hunk 5]
```diff
@@ -97,6 +106,14 @@
       }
     }
   }
+
+  &.pf-m-limit-width {
+    width: 100%;
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:111]: could this just be `max-width: 500px` with no need for a media query?

File: src/patternfly/components/Form/form.scss [hunk 6]
```diff
@@ -97,6 +106,14 @@
...
+
+    @media (min-width: $pf-global--breakpoint--md) {
+      width: var(--pf-c-form--m-limit-width--md--Width);
+    }
+  }
```
Comment (mattnolting) [inline src/patternfly/components/Form/form.scss:116]: If using an inline variable update, this block can go away.

File: src/patternfly/components/Form/form.scss [hunk 7]
```diff
@@ -76,6 +79,12 @@
       }
     }
 
+    .pf-c-form__group {
+      &.pf-m-width-100 {
+        display: block;
+      }
+    }
```
Comment (mattnolting) [inline src/patternfly/components/Form/form.scss:86]: Is this needed?

Comment (christiemolloy) [reply src/patternfly/components/Form/form.scss:86]: its to undo the display: grid thats set, should i do something else ?

File: src/patternfly/components/Form/form.scss [hunk 8]
```diff
@@ -108,6 +122,10 @@
 
 .pf-c-form__group-label {
   padding-bottom: var(--pf-c-form__group-label--PaddingBottom);
+
+  .pf-c-form__label + .pf-c-form__helper-text {
```
Comment (mcoker) [inline src/patternfly/components/Form/form.scss:126]: do you need this or could you just set `--pf-c-form__helper-text--MarginTop: 0;` on `.pf-c-form__group-label`?

**Review by mcoker — APPROVED** [review]
very nice!!

**Review by mcarrano — APPROVED** [review]
Looks great @christiemolloy !

**Pull request merged**